idl/paging_state.idl.hh

```diff
@@ -11,11 +11,13 @@ namespace pager {
 class paging_state {
     partition_key get_partition_key();
     std::optional<clustering_key> get_clustering_key();
-    uint32_t get_remaining();
+    uint32_t get_remaining_low_bits();
     utils::UUID get_query_uuid() [[version 2.2]] = utils::UUID();
     std::unordered_map<dht::token_range, std::vector<utils::UUID>> get_last_replicas() [[version 2.2]] = std::unordered_map<dht::token_range, std::vector<utils::UUID>>();
     std::optional<db::read_repair_decision> get_query_read_repair_decision() [[version 2.3]] = std::nullopt;
-    uint32_t get_rows_fetched_for_last_partition() [[version 3.1]] = 0;
+    uint32_t get_rows_fetched_for_last_partition_low_bits() [[version 3.1]] = 0;
+    std::optional<uint32_t> get_remaining_high_bits() [[version 4.3]];
+    std::optional<uint32_t> get_rows_fetched_for_last_partition_high_bits() [[version 4.3]];
```

Comment:
I wonder if we need the optional here. Won't those two fields default to zero?

Reply:
After reading the idl compiler it looks like you're right. For some reason I thought that the optionals were needed to generate the default initialization if the field can't be deserialized, while the deciding factor is actually a presence of [[version]] attribute.

Reply:
I am sorry you had to read the idl compiler source.